src/Microsoft.AspNet.OData.Shared/EdmEntityObject.cs

```diff
@@ -38,6 +40,31 @@ public EdmEntityObject(IEdmEntityTypeReference edmType)
         public EdmEntityObject(IEdmEntityType edmType, bool isNullable)
             : base(edmType, isNullable)
         {
+            TransientInstanceAnnotationContainer = new ODataInstanceAnnotationContainer();
+            PersistentInstanceAnnotationsContainer = new ODataInstanceAnnotationContainer();
+        }
+
+        /// <inheritdoc />
+        public IODataInstanceAnnotationContainer TransientInstanceAnnotationContainer { get; set; }
```

Comment:
>TransientInstanceAnnotationContainer [](start = 49, length = 36)

I need to think a bit more about how transient versus persistent instance annotations are exposed. initially I thought that the public interface would just have instance annotations and we would figure out which were transient and which were persistent. Persistent would be implemented by the <TStructuralType> (or EdmEntityType if there is no backing type) and transient would be implemented in a "wrapper".